### PR TITLE
FMWK-195 Override Set Name from Operations/Template

### PR DESCRIPTION
**Pull request by roimenashe**
Following https://github.com/aerospike/spring-data-aerospike/issues/454

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java [hunk 1]
```diff
@@ -1000,49 +1212,48 @@ private Record putAndGetHeader(AerospikeWriteData data, WritePolicy policy, bool
...
-
     @Override
     public <T> Stream<T> findAllUsingQuery(Class<T> entityClass, Filter filter,
                                            Qualifier qualifier) {
-        return findAllRecordsUsingQuery(entityClass, null, filter, qualifier)
-            .map(keyRecord -> mapToEntity(keyRecord.key, entityClass, keyRecord.record));
+        return findAllUsingQuery(entityClass, getSetName(entityClass), filter, qualifier);
     }
 
     public <T, S> Stream<?> findAllUsingQuery(Class<T> entityClass, Class<S> targetClass, Filter filter,
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java:1238]: * Should be added to the interface?
* Return `Stream<S>`
* Receive a single Qualifier (a common issue that have to be reviewed)

**Review by reugn — COMMENTED** [review]
Overall LGTM.

File: src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java (hunk 1 above)
Comment (roimenashe) [reply src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java:1238]: Can you explain?
We will need to refactor and expose only relevant "find" queries in the next PR but this specific line just uses different method internally.

Comment (reugn) [reply src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java:1238]: Yes, will be done in a separate PR.

**Review by agrgr — APPROVED** [review]

**Comment by roimenashe** [comment]
@reugn @agrgr do not merge yet, might add deprecated methods (with relevant message) to this PR just not to loose track of them. Deprecated methods drawback is that it conflicts with some of the new ones with method overloading [due to generics and raw Object usage, will investigate how to solve it if necessary.

**Pull request merged**